### PR TITLE
Update Connection State to Connection Status

### PR DESCRIPTION
**Pull request by logangingerich**
https://linear.app/workos/issue/SDK-117/add-connection-state-in-the-net-sdk

### CODE REVIEW

File: src/WorkOS.net/Services/SSO/Enums/ConnectionStatus.cs [hunk 1]
```diff
@@ -5,10 +5,10 @@
     using Newtonsoft.Json.Converters;
 
     /// <summary>
-    /// An enum describing the linked status of a <see cref="Connection"/>.
+    /// An enum describing the linked state of a <see cref="Connection"/>.
     /// </summary>
     [JsonConverter(typeof(StringEnumConverter))]
-    public enum ConnectionStatus
+    public enum ConnectionState
```
Comment (maxdeviant) [inline src/WorkOS.net/Services/SSO/Enums/ConnectionStatus.cs:11]: We need to change the values as well.

The possible values for `state` are `draft`, `active`, and `inactive`.

Also, I think we should go through a deprecation cycle for languages that support it.

So for .NET we should leave the existing `ConnectionStatus` enum and `Status` property on the `Connection` but mark them both with the [Obsolete attribute](https://docs.microsoft.com/en-us/dotnet/api/system.obsoleteattribute?view=net-5.0), with a message indicating that `State` should be used instead.

This gives developers a chance to transition without breaking the SDK usage immediately.

Comment (logangingerich) [reply src/WorkOS.net/Services/SSO/Enums/ConnectionStatus.cs:11]: gotcha

File: src/WorkOS.net/Services/SSO/Entities/Connection.cs [hunk 2]
```diff
@@ -19,11 +20,18 @@ public class Connection
...
+        [JsonProperty("state")]
+        public ConnectionState State { get; set; }
+
         /// <summary>
         /// The linked status of the Connection.
         /// </summary>
         [JsonProperty("status")]
-        public ConnectionStatus Status { get; set; }
+        [ObsoleteAttribute("The status property is obsolete. Please use state instead.", false)]
+        public ConnectionState Status { get; set; }
```
Comment (maxdeviant) [inline src/WorkOS.net/Services/SSO/Entities/Connection.cs:34]: The enum value here should be `ConnectionStatus`:

```suggestion
        public ConnectionStatus Status { get; set; }
```

File: src/WorkOS.net/Services/SSO/Entities/Connection.cs [hunk 3]
```diff
@@ -19,11 +20,18 @@ public class Connection
...
+        /// </summary>
+        [JsonProperty("state")]
+        public ConnectionState State { get; set; }
+
         /// <summary>
         /// The linked status of the Connection.
         /// </summary>
         [JsonProperty("status")]
-        public ConnectionStatus Status { get; set; }
+        [ObsoleteAttribute("The status property is obsolete. Please use state instead.", false)]
```
Comment (maxdeviant) [inline src/WorkOS.net/Services/SSO/Entities/Connection.cs:33]: Could we use the same casing as the property names in these deprecation messages?

```cs
[ObsoleteAttribute("The Status property is obsolete. Please use State instead.", false)]
```

**Review by maxdeviant — APPROVED** [review]

**Pull request merged**